libethashseal/Ethash.cpp

```diff
@@ -107,6 +107,12 @@ void Ethash::verify(Strictness _s, BlockHeader const& _bi, BlockHeader const& _p
 
 		if (_bi.number() && _bi.extraData().size() > chainParams().maximumExtraDataSize)
 			BOOST_THROW_EXCEPTION(ExtraDataTooBig() << RequirementError(bigint(chainParams().maximumExtraDataSize), bigint(_bi.extraData().size())) << errinfo_extraData(_bi.extraData()));
+
+		u256 daoHardfork = chainParams().u256Param("daoHardforkBlock");
+		bytes extraData = fromHex("0x64616f2d686172642d666f726b");
+		if (_bi.number() >= daoHardfork && _bi.number() <= daoHardfork + 9)
+			if (_bi.extraData() != extraData)
+				BOOST_THROW_EXCEPTION(ExtraDataIncorrect() << errinfo_comment("Block with this number require specific Extradata of 0x64616f2d686172642d666f726b!"));
```

Comment:
We could use a different error message that explains better what has happened:
`Received block from the wrong fork (invalid extradata).`